src/main/java/org/broadinstitute/hellbender/utils/recalibration/covariates/StandardCovariateList.java

```diff
@@ -119,11 +128,13 @@ public int indexByClass(final Class<? extends Covariate> clazz){
      * record the values in the provided storage object.
       */
     public void recordAllValuesInStorage(final GATKRead read, final SAMFileHeader header, final ReadCovariates resultsStorage) {
-        forEach(cov -> {
-            final int index = indexByClass(cov.getClass());
-            resultsStorage.setCovariateIndex(index);
+        //Note: extracting the field to a temp because this is a very tight loop and field lookup came up on the profile
+        final List<Covariate> allCovs = allCovariates;
```

Comment:
field lookup is slower than temp lookup. But I think this made marginal difference. Will use the field